Data/Aeson/Types/FromJSON.hs

```diff
@@ -178,11 +178,11 @@ parseRealFloat _        Null       = pure (0/0)
 parseRealFloat expected v          = typeMismatch expected v
 {-# INLINE parseRealFloat #-}
 
-truncateIntegral :: Integral a => String -> Scientific -> Parser a
-truncateIntegral expected s
-    | Scientific.isInteger s = pure (truncate s)
-    | otherwise               =
-        fail $ "Floating number specified for " ++ expected ++ ": " ++ show s
+truncateIntegral :: forall a. Integral a => String -> Scientific -> Parser a
```

Comment:
It's not exposed, so it's not that big deal. But will do if there are good versions. `parseIntegralFromScientific` (and `parseBoundedIntegralFromScientific`)?